include/openenclave/bits/seal.h

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/**
+ * @file seal.h
+ *
+ * This file defines constants and structures for sealing APIs.
+ *
+ * Only TEE agnostic definitions should go to this file. SGX specific
+ * definitions should go to sgx/seal.h
+ */
+
+#ifndef _OE_BITS_SEAL_H
+#define _OE_BITS_SEAL_H
+
+#include <openenclave/bits/evidence.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+typedef struct _oe_seal_setting_header
+{
+    int name;
+    size_t size;
+} oe_seal_setting_header_t;
+
+/**
+ * TEE agnostic seal settings supported by OE SDK
+ */
+typedef enum _oe_seal_setting_name
+{
+    OE_SEAL_SETTING_POLICY,
+    OE_SEAL_SETTING_ADDITIONAL_CONTEXT,
+    OE_SEAL_SETTING_TEE_AGNOSTIC_MAX
```

Comment:
```suggestion
```
This doesn't seem to be used anywhere and is often considered bad practice to define a MAX

Reply:
Per SIG-arch meeting, remove `OE_SEAL_SETTING_TEE_AGNOSTIC_MAX` and reserve at least 64K range of values reserved for OE SDK use as a constant.
- This range will be used for TEE agnostic and plugins shipping with OE SDK. 
- The implementation code should enforce that `type` provided in that range that is not currently defined by OE SDK is an error.